src/CoreWCF.Primitives/src/CoreWCF/IdentityModel/Tokens/SessionSecurityTokenHandler.cs

```diff
@@ -0,0 +1,733 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.IO;
+using System.Runtime.Serialization.Formatters.Binary;
+using System.Security.Claims;
+using System.Xml;
+using CoreWCF.IdentityModel.Claims;
+using CoreWCF.IdentityModel.Configuration;
+using CoreWCF.IdentityModel.Selectors;
+using CoreWCF.Runtime;
+using CoreWCF.Security;
+using SysUniqueId = System.Xml.UniqueId;
+
+namespace CoreWCF.IdentityModel.Tokens
+{
+    /// <summary>
+    /// A <see cref="SecurityTokenHandler"/> that processes <see cref="SessionSecurityToken"/>.
+    /// </summary>
+    public class SessionSecurityTokenHandler : SecurityTokenHandler
+    {
+        private const string DefaultCookieElementName = "Cookie";
+        private const string DefaultCookieNamespace = "http://schemas.microsoft.com/ws/2006/05/security";
+        private const string SecureConversationTokenIdentifier = "http://schemas.microsoft.com/ws/2006/05/servicemodel/tokens/SecureConversation";
+        public static readonly TimeSpan DefaultLifetime = TimeSpan.FromHours(10);
+        public static readonly ReadOnlyCollection<CookieTransform> DefaultCookieTransforms = (new List<CookieTransform>(new CookieTransform[] { new DeflateCookieTransform(), new ProtectedDataCookieTransform() }).AsReadOnly());
+        private TimeSpan _tokenLifetime = DefaultLifetime;
```

Comment:
ProtectedDataCookieTransform doesn't work on Linux. We should look to replace it with something based on the new asp.net core data protection api's which perform a very similar job. Details [here](https://docs.microsoft.com/en-us/aspnet/core/security/data-protection/introduction?view=aspnetcore-6.0). Only difficulty with the new api's is you have to get an implementation via DI.